README.md

```diff
@@ -61,90 +66,208 @@ SENTRY_DNS=8.8.8.8
 ```
+
 NB:
-* Ensure you use the prefix `REACT_APP_` to store all the variables that are needed in the React Application. Example: `REACT_APP_NAME=janedoe`
-* **DO NOT** commit any of the environment files to version control.
+
+- Ensure you use the prefix `REACT_APP_` to store all the variables that are needed in the React Application. Example: `REACT_APP_NAME=janedoe`
+- **DO NOT** commit any of the environment files to version control.
 
 ### Docker
 
 This application is built with Docker, you will need to have the following installed in order to use it:
-* `Makefile` - Use the link below to install https://makefiletutorial.com/.
-* `Docker` - https://www.docker.com/.
+
+- `Makefile` - Use the link below to install https://makefiletutorial.com/.
+- `Docker` - https://www.docker.com/.
+
+make sure you are in the root directory when running make commands
+
+### General
 
 1. Build Docker container and start service.
+   ```
+   make docker-build
+   ```
+2. Build Docker container in test environment
+   ```
+   make docker-build-test
+   ```
+3. Show logs from container processes.
+   ```
+   make show-logs
+   ```
+4. Stop container.
+
+   ```
+   make start
+   ```
+
+5. reset containers
+   ```
+   make reset
+   ```
+6. Pruning system images
+   ```
+   make prune
+   ```
+
+### Server commands
+
+4. Build Docker container and start service.
+   ```
+   make api-gem-setup
+   ```
+5. creating the database
+
+   ```
+   make api-db-create
+   ```
+
+6. Making database migration
+
+   ```
+   make api-db-migrate
+   ```
+
+7. Starting the server
+   ```
+   make api-start
+   ```
+8. Running backend linters tests
+
+   ```
+   make api-lint-fix
+   ```
+
+9. Running api tests
+
+   ```
+   make api-test
+   ```
+
+10. Starting the server
     ```
-    make docker-build 
+    make api-start
     ```
-2. Show logs from container processes.
+11. Running backend linters tests
+
     ```
-    make show-logs
+    make api-lint-fix
     ```
-3. Stop container.
+
+### Client
+
+10. Installing frontend dependancies
+
+    ```
+    make client-build
+    ```
+
+11. Installing npm packages in the client
+
     ```
-    make start 
+    make client-npm-setup
     ```
-4. All client sided routes are rendered from root, `/`. Example `http://localhost:8080/admin` - Admin React Page
-5. All API endpoints are rendered from `/api/`. Example `http://localhost:8080/api/admin` - Admin endpoint
+
+12. Starting the client
+
+    ```
+    make client-start
+    ```
+
+13. Client eslint checks
+
+    ```
+    make client-lint-check
+    ```
+
+14. Client eslint autofix
+
+    ```
+    make client-lint-fix
+    ```
+
+15. Client tests
+
+    ```
+    make client-test
+    ```
+
+16. All client sided routes are rendered from root, `/`. Example `http://localhost:5173/admin` - Admin React Page
+17. All API endpoints are rendered from `/api/`. Example `http://localhost:3000/api/admin` - Admin endpoint
 
 **NB: All requests are listening from port `8080`.**
 
```

Comment:
It is important to also indicate that:

- [ ] If one is using the docker environment, then they do not require any of the following: `gem setup`, `npm setup`, `backend server start` and `frontend server start`. They are all included done when container is built using `make docker-build`.
- [ ] All tests **must** be run in the `test` docker environment to avoid corrupting your local dev db.